pkg/monitortests/node/legacynodemonitortests/kubelet.go

```diff
@@ -238,64 +246,104 @@ func testContainerFailures(adminRestConfig *rest.Config, events monitorapi.Inter
+					excessiveExitsByNamespaceForFlakeTests[namespace] = append(excessiveExitsByNamespaceForFlakeTests[namespace], fmt.Sprintf("%s restarted %d times at:\n%s", locator, len(messages), strings.Join(messageSet.List(), "\n")))
+				}
 			}
 		}
 	}
-	sort.Strings(excessiveExits)
+	for namespace, excessiveExitsFails := range excessiveExitsByNamespaceForFailedTests {
+		sort.Strings(excessiveExitsFails)
+		excessiveExitsByNamespaceForFailedTests[namespace] = excessiveExitsFails
+	}
+	for namespace, excessiveExitsFlakes := range excessiveExitsByNamespaceForFlakeTests {
+		sort.Strings(excessiveExitsFlakes)
+		excessiveExitsByNamespaceForFlakeTests[namespace] = excessiveExitsFlakes
+	}
 
 	var testCases []*junitapi.JUnitTestCase
 
-	const failToStartTestName = "[sig-architecture] platform pods should not fail to start"
-	if len(failures) > 0 {
-		testCases = append(testCases, &junitapi.JUnitTestCase{
-			Name:      failToStartTestName,
-			SystemOut: strings.Join(failures, "\n"),
-			FailureOutput: &junitapi.FailureOutput{
-				Output: fmt.Sprintf("%d container starts had issues\n\n%s", len(failures), strings.Join(failures, "\n")),
-			},
-		})
+	for _, namespace := range sets.List(openshiftNamespaces) { // this ensures we create test case for every namespace, even in success cases
+		failures := failuresByNamespace[namespace]
+		failToStartTestName := fmt.Sprintf("[sig-architecture] platform pods in ns/%s should not fail to start", namespace)
+		if len(failures) > 0 {
+			testCases = append(testCases, &junitapi.JUnitTestCase{
+				Name:      failToStartTestName,
+				SystemOut: strings.Join(failures, "\n"),
+				FailureOutput: &junitapi.FailureOutput{
+					Output: fmt.Sprintf("%d container starts had issues\n\n%s", len(failures), strings.Join(failures, "\n")),
+				},
+			})
+		}
+		// mark flaky for now while we debug
+		testCases = append(testCases, &junitapi.JUnitTestCase{Name: failToStartTestName})
 	}
-	// mark flaky for now while we debug
-	testCases = append(testCases, &junitapi.JUnitTestCase{Name: failToStartTestName})
 
 	// We want to deflake this test.
 	// Plan is to release this test and report any failures for pods
 	// that restart more than 3 times
 	// We will then build exclusion rules for those that we see
 	// and then make this test fail for any case that doesn't match the rules
 	// we have.
-	const excessiveRestartTestName = "[sig-architecture] platform pods should not exit more than once with a non-zero exit code"
-	if len(excessiveExits) > 0 {
-		testCases = append(testCases, &junitapi.JUnitTestCase{
-			Name:      excessiveRestartTestName,
-			SystemOut: strings.Join(excessiveExits, "\n"),
-			FailureOutput: &junitapi.FailureOutput{
-				Output: fmt.Sprintf("%d containers with multiple restarts\n\n%s", len(excessiveExits), strings.Join(excessiveExits, "\n\n")),
-			},
-		})
+	for _, namespace := range sets.List(openshiftNamespaces) { // this ensures we create test case for every namespace, even in success cases
```

Comment:
This doesn't seem to be true, it looks like the test is only created on failure

Reply:
Apologies but I'm not seeing where you are creating a success case for this test:

`fmt.Sprintf("[sig-architecture] platform pods in ns/%s should not exit more than %d with a non-zero exit code", namespace, maxRestartCountForFailures)`

Every unique test name [needs to have a result in every job](https://github.com/openshift-eng/ci-test-mapping?tab=readme-ov-file#test-sources).

Reply:
I don't really understand this.

If I report both a success and a failure then the test is considered flaky. So I only want to report failures in this case.

Are you saying that every namespace in openshift should have a success job for this unless it fails?

cc @deads2k

Reply:
Yes, every namespace in openshift should have a successful result for `[sig-architecture] platform pods in ns/%s should not exit more than %d with a non-zero exit code`, unless it fails.

Reply:
How would you otherwise know the pass rate of a test unless you collect both successes and failures?

Reply:
https://github.com/openshift/origin/pull/29118

Reply:
Thanks, that fixes the success case.  

We also compare test histories across releases, so having thresholds (3 and 2) in the names is problematic (as well as the "flake" part), since it causes the names to change if you adjust them.

I think you could call the test `[sig-architecture] platform pods in ns/%s should not exit excessively with a non-zero exit code`-- and in the failure message mention what the current excessive limit is.  It also lets you have per-operator limits (e.g. limit OLM to 6 restarts, so it can't get any worse) and let's you ratchet the values down without changing the names.

I'm not sure if the flake thresholds are meant to truly be a separate test either, generally you could use 1 test name for everything and decide whether it should fail, flake, or succeed.